### PR TITLE
Defer execution of promises for playSpeech block

### PR DESCRIPTION
**Pull request by maddiedierker**
Fixes a bug where promises were automatically executed when the playSpeech block was called. Execution should be deferred to take advantage of caching when duplicate requests are triggered in quick succession.

More explicitly, if a student had this code in Gamelab:
```
function draw() {
  playSpeech("hi", "female", "English");
}
```

Many network requests would be triggered instead of just 1 before client caching kicked in (enough for us to get rate-limited almost immediately). This is because `createSoundPromise` returned a promise that was immediately executed. Now, `createSoundPromise` returns a thunk so that each promise in the queue is executed synchronously.

## Follow-up Work

I did some basic renaming + documentation updates in this PR, but I'd like to come back later to clean up my naming conventions here while keeping this PR small.

## Testing story

Updates unit tests.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/AzureTextToSpeech.js [hunk 1]
```diff
@@ -74,9 +73,11 @@ export default class AzureTextToSpeech {
    * @param {string} opts.ssml SSML in request body.
    * @param {string} opts.token Authentication token from Azure.
    * @param {function(Array<string>)} opts.onProfanityFound Called if the given text contains profanity.
-   * @returns {Promise<SoundResponse>} A promise that returns a SoundResponse when resolved.
+   * @returns {function} A thunk that returns promise, which resolves to a SoundResponse. Example usage:
+   * const soundPromise = createSoundPromise(options);
+   * const soundResponse = await soundPromise();
    */
-  createSoundPromise = opts => {
+  createSoundPromise = opts => () => {
```
Comment (maddiedierker) [inline apps/src/AzureTextToSpeech.js:80]: this is the actual change 😄  everything else in this PR is updating consumers, documentation, and unit tests

File: apps/src/AzureTextToSpeech.js [hunk 2]
```diff
@@ -224,17 +225,17 @@ export default class AzureTextToSpeech {
   };
 
   /**
-   * Add a promise to the end of the queue.
-   * @param {Promise<SoundResponse>} promise A promise that returns a SoundResponse when resolved.
+   * Add to the end of the queue.
+   * @param {function} promise A thunk that returns promise, which resolves to a SoundResponse.
```
Comment (ajpal) [inline apps/src/AzureTextToSpeech.js]: super nit: should be "A thunk that returns a promise"

File: apps/src/AzureTextToSpeech.js [hunk 3]
```diff
@@ -55,16 +55,15 @@ export default class AzureTextToSpeech {
 
   /**
    *
-   * @param {Promise<SoundResponse>} soundPromise A promise that returns a SoundResponse when resolved.
-   * playback configuration options.
+   * @param {function} soundPromise A thunk that returns promise, which resolves to a SoundResponse.
```
Comment (ajpal) [inline apps/src/AzureTextToSpeech.js]: super nit: should be "returns a promise"

File: apps/src/AzureTextToSpeech.js (hunk 1 above)
Comment (ajpal) [reply apps/src/AzureTextToSpeech.js:80]: thank you for this comment!! made this much easier to review :)

**Review by ajpal — APPROVED** [review]

**Pull request merged**